resource-aggregate/events/resourceLinksSnapshotTaken.go

```diff
@@ -45,7 +45,35 @@ func (e *ResourceLinksSnapshotTaken) IsSnapshot() bool {
 	return true
 }
 
-func (e *ResourceLinksSnapshotTaken) HandleEventResourceLinksPublished(ctx context.Context, pub *ResourceLinksPublished) error {
+// Check if the snapshot of published resources matches exactly the new published data
+func (e *ResourceLinksSnapshotTaken) MatchesLinksPublished(published *ResourceLinksPublished) bool {
+	if published == nil {
+		return false
+	}
+
+	if len(e.Resources) != len(published.Resources) {
+		return false
+	}
+
+	for _, resPub := range published.GetResources() {
+		resSnap, ok := e.Resources[resPub.Href]
+		if !ok {
+			return false
```

Comment:
10 resources {a,b,c...j} are already published. 5 resources are being published by the device after one hour because user was playing with the device when it was disconnected from the cloud. These 5 resources which are contained in the publish request are {a,b,x,y,z}.
What happens is (assuming `if len(e.Resources) != len(published.Resources)` is removed) based on the order, either user is notified about all 5 resources or not at all.

This function shouldn't return true/false, but it should return diff of resources which are not yet published - which are not yet part of the `e`.

The same is done in the `HandleEventResourceLinksUnpublished`, which notifies user only about those unpublished resources, which were really unpublished.

Reply:
Reworked to return slice of resources not yet cached and updated resources